apps/jobs/views.py

```diff
@@ -2073,3 +2074,31 @@ def get_github_badge_data(
         else:
             data["message"] = f"{challenge_obj.title}"
     return Response(data, status=http_status_code)
+
+
+@api_view(["GET"])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def challenge_phase_submissions_by_status(request, challenge_phase_id):
+    """
```

Comment:
Can we please add all the checks here so that this API can only be accessed by challenge host.